homeassistant/components/homekit/thermostats.py

```diff
@@ -0,0 +1,214 @@
+"""Class to hold all thermostat accessories."""
+import logging
+
+from homeassistant.components.climate import (
+    ATTR_CURRENT_TEMPERATURE, ATTR_TEMPERATURE,
+    ATTR_TARGET_TEMP_HIGH, ATTR_TARGET_TEMP_LOW,
+    ATTR_OPERATION_MODE, STATE_HEAT, STATE_COOL,
+    STATE_AUTO, SUPPORT_TARGET_TEMPERATURE_HIGH,
+    SUPPORT_TARGET_TEMPERATURE_LOW)
+from homeassistant.const import (
+    ATTR_ENTITY_ID, ATTR_UNIT_OF_MEASUREMENT,
+    ATTR_SUPPORTED_FEATURES, TEMP_CELSIUS, TEMP_FAHRENHEIT)
+from homeassistant.helpers.event import async_track_state_change
+
+from . import TYPES
+from .accessories import HomeAccessory, add_preload_service
+from .const import (
+    SERV_THERMOSTAT, CHAR_CURRENT_HEATING_COOLING,
+    CHAR_TARGET_HEATING_COOLING, CHAR_CURRENT_TEMPERATURE,
+    CHAR_TARGET_TEMPERATURE, CHAR_TEMP_DISPLAY_UNITS,
+    CHAR_COOLING_THRESHOLD_TEMPERATURE, CHAR_HEATING_THRESHOLD_TEMPERATURE)
+
+_LOGGER = logging.getLogger(__name__)
+
+UNIT_HASS_TO_HOMEKIT = {TEMP_CELSIUS: 0, TEMP_FAHRENHEIT: 1}
+UNIT_HOMEKIT_TO_HASS = {c: s for s, c in UNIT_HASS_TO_HOMEKIT.items()}
+THC_HASS_TO_HOMEKIT = {'off': 0, STATE_HEAT: 1, STATE_COOL: 2, STATE_AUTO: 3}
+THC_HOMEKIT_TO_HASS = {c: s for s, c in THC_HASS_TO_HOMEKIT.items()}
+CHC_HASS_TO_HOMEKIT = {'off': 0, STATE_HEAT: 1, STATE_COOL: 2}
+CHC_HOMEKIT_TO_HASS = {c: s for s, c in CHC_HASS_TO_HOMEKIT.items()}
+
+
+@TYPES.register('Thermostat')
+class Thermostat(HomeAccessory):
+    """Generate a Thermostat accessory for an alarm control panel."""
+
+    def __init__(self, hass, entity_id, display_name):
+        """Initialize a Thermostat accessory object."""
+        super().__init__(display_name, entity_id, 'THERMOSTAT')
+
+        self._hass = hass
+        self._entity_id = entity_id
+        self._call_timer = None
+
+        self.heat_cool_flag_target_state = False
+        self.temperature_flag_target_state = False
+        self.cooling_thresh_temp_flag_target_state = False
+        self.heating_thresh_temp_flag_target_state = False
+
+        self.service_thermostat = add_preload_service(self, SERV_THERMOSTAT)
```

Comment:
You would need to add the optional characteristics to add to the service as a list here. Otherwise they wouldn't be initiated for the accessory.
```py
add_preload_service(self, SERV_THERMOSTAT, [CHAR_COOLING_THRESHOLD_TEMPERATURE, CHAR_HEATING_THRESHOLD_TEMPERATURE])
```